examples/shouldi/shouldi/dependency_check.py

```diff
@@ -0,0 +1,53 @@
+import json
+import asyncio
+from typing import Dict, Any
+
+from dffml.df.base import op
+from dffml.df.types import Definition
+
+package_src_dir = Definition(name="package_src_dir", primitive="str")
+dependency_check_output = Definition(
+    name="dependency_check_output", primitive="Dict[str, Any]"
+)
+
+
+class DependencyCheckError(Exception):
+    """
+    Raised when dependency-check fails
+    """
+
+
+@op(
+    inputs={"pkg": package_src_dir},
+    outputs={"report": dependency_check_output},
+)
+async def run_dependency_check(pkg: str) -> Dict[str, Any]:
+    """
+    CLI usage: dffml service dev run -log debug shouldi.dependency_check:run_dependency_check -pkg .
+    """
+    proc = await asyncio.create_subprocess_exec(
+        "dependency-check",
```

Comment:
If you look in `"dependency-check/bin` you'll see

```console
$ ls -lAF dependency-check/bin/
total 8
-rw-r--r-- 1 pdxjohnny pdxjohnny 3003 Mar 26 09:04 dependency-check.bat
-rwxr-xr-x 1 pdxjohnny pdxjohnny 3527 Mar 26 09:04 dependency-check.sh*
```

But no `dependency-check`. `.bat` is a a windows / MS-DOS script. `.sh` is a bash (Linux/OSX) script, so we want to run that one.

```suggestion
        "dependency-check.sh",
```

Reply:
when it tries to reach `dependency_check.sh`, it shows `permission denied`

Reply:
`dependency_check.sh`, or `dependency-check.sh`?

Reply:
Always push up the code in question so we can read the GitHub actions to see why it's failing :)

Reply:
okay, will remember in future .. 
Thanks :)